src/video/SDL_video.c

```diff
@@ -630,10 +630,21 @@ SDL_DisplayID SDL_AddVideoDisplay(const SDL_VideoDisplay *display, SDL_bool send
     SDL_VideoDisplay *displays, *new_display;
     SDL_DisplayID id = 0;
 
-    displays = (SDL_VideoDisplay *)SDL_realloc(_this->displays, (_this->num_displays + 1) * sizeof(*displays));
+    displays = (SDL_VideoDisplay *)SDL_malloc((_this->num_displays + 1) * sizeof(*displays));
     if (displays) {
         int i;
+
+        /* The display list may contain self-referential pointers to the desktop mode. */
+        SDL_memcpy(displays, _this->displays, _this->num_displays * sizeof(*displays));
+        for (i = 0; i < _this->num_displays; ++i) {
+            if (displays[i].current_mode == &_this->displays[i].desktop_mode) {
+                displays[i].current_mode = &displays[i].desktop_mode;
+            }
```

Comment:
Also need to check for current mode being a fullscreen mode here.

Reply:
The fullscreen list is a separately allocated array, so it remains static even when reallocating the display list due to adding a display. When resizing the display array, the pointer to the fullscreen mode list is just copied over without changing the addresses of the elements, so if the current mode points to a fullscreen mode here, it will still point to valid memory.

Reply:
Ah, good point, thanks!